### PR TITLE
Check vite.emptyOutDir before emptying dir

### PR DESCRIPTION
**Pull request by torchsmith**
## Changes

- What does this change?

Adds `vite.build.emptyOutDir` functionality

Not sure how to use pnpm or changeset.

## Testing

No tests. Not sure how to add a test. Not sure how to build astro and manually test locally etc.

## Docs

No Docs needed because this fits into existing docs.
Vite config should work within Astro config already.

### CODE REVIEW

**Comment by torchsmith** [comment]
https://github.com/withastro/astro/issues/6151

**Review by natemoo-re — CHANGES_REQUESTED** [review]

File: packages/astro/src/core/build/static-build.ts [hunk 1]
```diff
@@ -61,7 +61,9 @@ export async function staticBuild(opts: StaticBuildOptions) {
 	// Empty out the dist folder, if needed. Vite has a config for doing this
 	// but because we are running 2 vite builds in parallel, that would cause a race
 	// condition, so we are doing it ourselves
-	emptyDir(settings.config.outDir, new Set('.git'));
+	if(!settings.config?.vite?.build?.emptyOutDir) {
```
[the remark this answers is not in the document]
Comment (torchsmith) [reply packages/astro/src/core/build/static-build.ts:64]: Right, good catch! My bad.

**Review by natemoo-re — APPROVED** [review]

**Comment by torchsmith** [comment]
Im not sure what i need to do here, if anything. Just let me know what I can do to get this merged into next patch.

**Comment by natemoo-re** [comment]
Thanks @torchsmith! Nothing to do, I'm just waiting for CI to pass and then will merge.

Appreciate the fix!

**Pull request merged**